Mathlib/Algebra/Group/Action/Basic.lean

```diff
@@ -146,3 +146,55 @@ variable [Group α] [Monoid β] [MulAction α β] [SMulCommClass α β β] [IsSc
   ⟨fun h => inv_smul_smul g m ▸ h.smul g⁻¹, IsUnit.smul g⟩
 
 end SMul
+
+section MulDistribMulAction
+variable [Monoid M] [Monoid A] [MulDistribMulAction M A]
+
+/-- Pullback a multiplicative distributive multiplicative action along an injective monoid
+homomorphism. -/
+-- See note [reducible non-instances]
+protected abbrev Function.Injective.mulDistribMulAction [Monoid B] [SMul M B] (f : B →* A)
+    (hf : Injective f) (smul : ∀ (c : M) (x), f (c • x) = c • f x) : MulDistribMulAction M B where
+  __ := hf.mulAction f smul
+  smul_mul c x y := hf <| by simp only [smul, f.map_mul, smul_mul']
+  smul_one c := hf <| by simp only [smul, f.map_one, smul_one]
+
+/-- Pushforward a multiplicative distributive multiplicative action along a surjective monoid
+homomorphism. -/
+-- See note [reducible non-instances]
+protected abbrev Function.Surjective.mulDistribMulAction [Monoid B] [SMul M B] (f : A →* B)
+    (hf : Surjective f) (smul : ∀ (c : M) (x), f (c • x) = c • f x) : MulDistribMulAction M B where
+  __ := hf.mulAction f smul
+  smul_mul c := by simp only [hf.forall, smul_mul', ← smul, ← f.map_mul, implies_true]
+  smul_one c := by rw [← f.map_one, ← smul, smul_one]
+
+variable (A) in
+/-- Scalar multiplication by `r` as a `MonoidHom`. -/
+@[simps] def MulDistribMulAction.toMonoidHom (r : M) : A →* A where
```

Comment:
I assume you checked that `@[simps]` generates the same lemma as the previous `MulDistribMulAction.toMonoidHom_apply`?